Scripts/Runtime/Entities/Tasks/AbstractTaskDriverSystem.cs

```diff
@@ -12,89 +12,139 @@ namespace Anvil.Unity.DOTS.Entities
     /// </summary>
     public abstract partial class AbstractTaskDriverSystem : AbstractAnvilSystemBase
     {
+        private const int SYSTEM_LEVEL_CONTEXT = -2;
+
         private readonly List<AbstractTaskDriver> m_TaskDrivers;
         private readonly VirtualDataLookup m_InstanceDataLookup;
         private readonly List<JobTaskWorkConfig> m_UpdateJobData;
+        private readonly List<JobTaskWorkConfig> m_HandleCancelledJobData;
+
+        private int m_TaskDriverID;
+
+#if ENABLE_UNITY_COLLECTIONS_CHECKS
+        private bool m_HasCheckedUpdateJobsForDataLoss;
+#endif
+
+        internal CancelData CancelData
+        {
+            get;
+        }
+
         protected AbstractTaskDriverSystem()
         {
             m_InstanceDataLookup = new VirtualDataLookup();
+            CancelData = new CancelData(this);
+
             m_TaskDrivers = new List<AbstractTaskDriver>();
             m_UpdateJobData = new List<JobTaskWorkConfig>();
+            m_HandleCancelledJobData = new List<JobTaskWorkConfig>();
         }
-        
+
         protected override void OnDestroy()
         {
             m_InstanceDataLookup.Dispose();
-            
+            CancelData.Dispose();
+
             m_UpdateJobData.Clear();
-            
+            m_HandleCancelledJobData.Clear();
+
             //Note: We don't dispose TaskDrivers here because their parent or direct reference will do so. 
             m_TaskDrivers.Clear();
 
             base.OnDestroy();
         }
 
+        internal int GetNextID()
+        {
+            //TODO: DISCUSS - Make this robust
```

Comment:
While very unlikely since this ID represents over 2 billion TaskDrivers tied to a specific System and TaskDrivers don't really get created/destroyed at runtime.... there's still a tiny chance of overflow. 

Should be bother making an IDProvider that keeps track of open ids? Or just keep like so and add an issue to make one in the future?

Reply:
Fine for now, make a TODO but

 - Make sure an overflow exception will actually get thrown so we catch this in the moment not as weird behaviour emerges. I DO NOT want to debug that!
 - If there is no overflow exception then throw your own exception
 - I don't see where `m_TaskDriverID` is getting initialized but shouldn't you be using `uint` and initializing to 0?

In a pinch we can always kick the can down the road and switch to `long` ids 😆

Reply:
Cool, sounds good.